lib/recurly/element/element.js

```diff
@@ -0,0 +1,389 @@
+    const container = find(maybes, maybe => INSTANCE_REF_NAME in maybe);
+    if (container) return container[INSTANCE_REF_NAME];
+  }
+
+  constructor ({ elements, ...options }) {
+    super();
+    this.elements = elements;
+    this.recurly = elements.recurly;
+    this.bus = elements.bus;
+    this.id = uid();
+    this._config = {};
+    this.state = {};
+
+    this.configure(options);
+    this.bus.add(this);
+    this.elements.add(this);
+
+    this.on(this.messageName('state:change'), (...args) => this.onStateChange(...args));
+    this.on(this.messageName('focus'), () => this.onFocus());
+    this.on(this.messageName('blur'), () => this.onBlur());
+    this.on(this.messageName('tab:previous'), () => this.onTab('previous'));
+    this.on(this.messageName('tab:next'), () => this.onTab('next'));
+    this.on(this.messageName('submit'), () => this.onSubmit());
+    this.on('destroy', (...args) => this.destroy(...args));
+    debug('create', this.id);
+  }
+
+  /**
+   * @private
+   */
+  get type () {
+    return this.constructor.type;
+  }
+
+  get supportsTokenization () {
+    return this.constructor.supportsTokenization;
+  }
+
+  get elementClassName () {
+    return this.constructor.elementClassName;
+  }
+
+  /**
+   * @private
+   */
+  get container () {
+    if (this._container) return this._container;
+
+    const { classList, iframe, tabProxy, id } = this;
+    const { DATA_ATTRIBUTE_ID, INSTANCE_REF_NAME } = this.constructor;
+
+    const container = this._container = document.createElement('div');
+    dom.data(container, DATA_ATTRIBUTE_ID, id);
+    container[INSTANCE_REF_NAME] = this;
+    container.setAttribute('class', classList);
+    container.appendChild(tabProxy);
+    container.appendChild(iframe);
+    container.addEventListener('click', () => this.focus());
+
+    return container;
+  }
+
+  /**
+   * @private
+   */
+  get tabProxy () {
+    if (this._tabProxy) return this._tabProxy;
+    const proxy = this._tabProxy = dom.createHiddenInput();
+    proxy.addEventListener('focus', () => this.focus());
+    return proxy;
+  }
+
+  /**
+   * @private
+   */
+  get iframe () {
+    if (this._iframe) return this._iframe;
+    const { id, url } = this;
+    const { FRAME_NAME_PREFIX } = this.constructor;
+    const iframe = this._iframe = document.createElement('iframe');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('scrolling', 'no');
+    iframe.setAttribute('name', `${FRAME_NAME_PREFIX}${id}`);
+    iframe.setAttribute('allowpaymentrequest', 'true');
+    iframe.setAttribute('style', 'background: none; width: 100%; height: 100%;');
+    iframe.src = url;
+    return iframe;
+  }
+
+  /**
+   * @private
+   */
+  get window () {
+    return this.iframe.contentWindow;
+  }
+
+  /**
+   * @private
+   *
+   * FIXME: contentWindow access is very slow.
+   *   - we may wish to use a sentinel to track attached state,
```

Comment:
Any thoughts on making a ticket to pick this up at some point?

Reply:
Yeah that would be useful, certainly. It's only potentially perceivable in IE11 when making repeated configure calls